chain/db/store.go

```diff
@@ -264,12 +264,12 @@ func (cs *ChainStore) persist(b *Block) error {
 			return err
 		}
 
-		if txn.UnsignedTx.Payload.Type != CoinbaseType && txn.UnsignedTx.Payload.Type != CommitType {
-			pg, _ := ToCodeHash(txn.Programs[0].Code)
-			acc := states.GetOrNewAccount(pg)
+		if txn.UnsignedTx.Payload.Type != CommitType {
+			pg, _ := txn.GetProgramHashes()
+			acc := states.GetOrNewAccount(pg[0])
```

Comment:
Can we guarantee len(pg) > 0?

Reply:
pg is exist certainly, because this block has passed block verification.